### PR TITLE
Auto api

### LINKED ISSUE #829: ERT should automatically start an api server
Our users should not have to manage the life cycle of the api server.

### PR DESCRIPTION
**Pull request by jondequinor**
The `gui` subcommand now supports the following arguments:
- `--storage-api-url STORAGE_API_URL` which will connect the `StorageClient` to this URL, and
- omitting the former argument or providing `--storage-api-bind STORAGE_API_BIND`, in either case will make ERT start its own Storage API server binding `127.0.0.0:0` or the provided parameter.

The implementation is ERT providing the Flask dev server with a socket. All of this is encapsulated in the `AutoClient` which is obtained using `create_client`.

The default values of the `StorageClient` `BASE_URL` has been moved to the `ert_api` argparser in the `bind` argument.

So starting api on a non-default address, requires you to pass `--bind 0.0.0.0:1337` to the `ert api` subcommand.

Closes #829

### CODE REVIEW

File: tests/storage/test_storage_client.py [hunk 1]
```diff
@@ -106,7 +113,7 @@ def test_observation_values(test_client):
                     datetime.strptime("2000-01-01 20:01:01", format),
                     datetime.strptime("2000-01-02 20:01:01", format),
                 ],
-                [4, 5]
+                [4, 5],
```
Comment (ManInFez) [inline tests/storage/test_storage_client.py:116]: Why the extra comma?

Comment (jondequinor) [reply tests/storage/test_storage_client.py:116]: Black I assume

Comment (ManInFez) [reply tests/storage/test_storage_client.py:116]: 👍

**Comment by ManInFez** [comment]
Is it a good thing to pass around the args object around, instead of just giving the vital information as separate parameters?

File: ert_shared/storage/autoclient.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class AutoClient(StorageClient):
+    """A client that will start its own server, and tears it down on request.
+    """
+
+    def __init__(self, bind):
+        StorageClient.__init__(self, "")
+
+        bind_parts = bind.split(":")
```
Comment (ManInFez) [inline ert_shared/storage/autoclient.py:19]: renamed to the purpose of the two separate items [ip_address, port]

File: ert_shared/storage/autoclient.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+        logging.info("Serving Storage API on {}".format(self._BASE_URI))
+
+        # XXX: a hack to get flask to pick up our created socket, for more
+        # serious application servers, this would be passed explicitly to the
+        # server interface.
+        os.environ["WERKZEUG_SERVER_FD"] = str(sock.fileno())
+
+        args = _Namespace(bind="{}:{}".format(address, port), debug=False)
+        self._thread = Process(target=run_server, args=(args,))
```
Comment (ManInFez) [inline ert_shared/storage/autoclient.py:36]: Why are we not using subprocess?

**Pull request merged**